apps/webapp/app/swap/swap-form/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { Button } from 'ui';
+import { FilledImage } from '../../../shared';
+import { useStore } from '../../../state';
+import { SwapToken, swapSelector } from '../../../state/swap';
+import { validateAmount } from '../../../utils';
+import { SwapInput } from './swap-input';
+
+export const SwapForm = () => {
+  const { pay, receive, validationErrors, setAmount, setAsset, replaceAsset } =
+    useStore(swapSelector);
+
+  return (
+    <form
+      className='flex flex-col gap-10'
+      onSubmit={e => {
+        e.preventDefault();
+      }}
+    >
+      <div className='relative flex flex-col gap-2'>
+        <SwapInput
+          placeholder='Enter an amount'
+          asset={pay.asset}
+          setAsset={setAsset(SwapToken.PAY)}
+          value={pay.amount}
```

Comment:
Currently not able to enter in two amounts:
https://github.com/penumbra-zone/web/assets/16624263/5d81f793-57da-46ee-a31a-684b788a997e

Reply:
this is currently a feature) ideally, the pay field should change automatically if the user changes the receive field and vice versa! so I'm still cleaning it up

Reply:
will we be able to display the pay value when the user changes the receive value?

Reply:
Ah! That's right. Do you know if penumbra supports both exact out and exact in? Forgot about these. 🤔 

If we only support exact in, then we should not allow users to edit the second box at all.

Reply:
https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.core.component.dex.v1alpha1#penumbra.core.component.dex.v1alpha1.SimulationService.SimulateTrade - it returns the value how much we can receive, but we can also get how much we need to pay, right?

Reply:
That simulation endpoint implies we only support _exact in_. This means we provide the asset id + amount as input and just the asset id as an output. It then returns how much we can expect to receive.

Reply:
if we pass the value from the second field to the input and the value from the first field to the output?Will it not work?

Reply:
1 penumbra = 15 ugm
15 ugm = 1 penumbra
is it right?

Reply:
> 1 penumbra = 15 ugm
This will be a dynamic number and always changing given market conditions.

Reply:
I gave this as an example

Reply:
> if we pass the value from the second field to the input and the value from the first field to the output?Will it not work?

Not sure exactly sure what you mean, but if we only allow exact in, this means when we click that arrow, the amount should stay in the first input and not move to the second. We should not be able to edit the second input.

Reply:
<img width="478" alt="Знімок екрана 2023-10-06 о 15 33 33" src="https://github.com/penumbra-zone/web/assets/82594516/9a851f8a-e488-4179-834b-bd546c380e48">

user input pay : input - firtst input value and output - second input value

<img width="475" alt="Знімок екрана 2023-10-06 о 15 35 11" src="https://github.com/penumbra-zone/web/assets/82594516/3907fc04-bced-430c-aa4c-f1427362a2ca">

user input receive: input - second input value and output - first input value
<img width="479" alt="Знімок екрана 2023-10-06 о 15 35 32" src="https://github.com/penumbra-zone/web/assets/82594516/ef7d6f01-951a-452a-b8f0-421835a960de">

Reply:
Will it work?

Reply:
@erwanor need some assistance here. Do I understand this right?

[This protobuf definition](https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.core.component.dex.v1alpha1#penumbra.core.component.dex.v1alpha1.SimulationService.SimulateTrade) says that you need to provide:
- input amount
- input asset id
- output asset id

And you'll receive a [SwapExecution](https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.core.component.dex.v1alpha1#penumbra.core.component.dex.v1alpha1.SwapExecution).

This will include the original input amount (right? the one we added ourselves) but now also include the output amount. Also, Penumbra only supports exact in right?

Reply:
@grod220 that's exactly it, and we only have support for exact in, though it's possible to build an exact out api on top of `SimulateTrade`. I can detail how async.